src/NServiceBus.Core/Serialization/SerializationConfigExtensions.cs

```diff
@@ -67,14 +67,31 @@ public static SerializationExtensions<T> AddDeserializer<T>(this EndpointConfigu
 
             var settings = new SettingsHolder();
             additionalSerializers.Add(Tuple.Create<SerializationDefinition, SettingsHolder>(serializationDefinition, settings));
-            return CreateSerializationExtension<T>(settings);
+            return CreateSerializationExtension<T>(settings, config.Settings);
         }
 
-        static SerializationExtensions<T> CreateSerializationExtension<T>(SettingsHolder settings) where T : SerializationDefinition
+        /// <summary>
+        /// Disables inference of message type based on the content type if the message type can't be determined by the 'NServiceBus.EnclosedMessageTypes' header.
+        /// </summary>
+        public static void DisableMessageTypeInference<T>(this SerializationExtensions<T> config) where T : SerializationDefinition
+        {
+            Guard.AgainstNull(nameof(config), config);
+
+            config.EndpointConfigurationSettings.Set(DisableMessageTypeInferenceKey, true);
+        }
+
+        internal static bool IsMessageTypeInferenceDisabled(this IReadOnlySettings endpointConfigurationSettings)
+        {
+            return endpointConfigurationSettings.GetOrDefault<bool>(DisableMessageTypeInferenceKey);
+        }
```

Comment:
In the other PR I've inversed this internal API. From a consuming perspective a bit more natural.

```suggestion
        internal static bool IsMessageTypeInferenceEnabled(this IReadOnlySettings endpointConfigurationSettings)
        {
            return !endpointConfigurationSettings.GetOrDefault<bool>(DisableMessageTypeInferenceKey);
        }
```

Reply:
I like it